packages/headless/src/api/search/search-api-client.ts

```diff
@@ -191,6 +190,22 @@ export class SearchAPIClient {
       error: unwrapError(platformResponse),
     };
   }
+
+  private getAbortControllerInstanceIfAvailable(): AbortController | null {
+    // For nodejs environments only, we want to load the implement of AbortController
+    // For browser environments, we need to make sure that we don't use AbortController as it might not be available (Locker Service in Salesforce)
+    // And since this cannot be polyfilled in a meaningful manner (this is a low level browser API after all, only JS code cannot cancel network requests)
+    // It is better to simply do nothing.
```

Comment:
If we are removing the behaviour from the browser, could we remove it all together and remove this comment? Is there a meaningful benefit of keeping the logic only when running in the server?

Reply:
We are not removing it from the browser.
We are removing it when when the browser either does not support it, or in the case of locker service, when the object is not proxied.

Reply:
I see, I focused on this part of the comment instead of the code:

```
For browser environments, we need to make sure that we don't use AbortController
```

It's not clear to me what are the benefits of AbortController are? Is it conserving QPM? My understanding is that firing a request is synchronous, so aborting will always target a request that is already in-flight?

I feel there is an unverified assumption, and we are adding more code around it for Lockerservice instead of quantifying the benefit of the code in the first place.